yoyodyne/models/lstm.py

```diff
@@ -239,21 +242,27 @@ def decode(
                 decoder_input, decoder_hiddens, encoder_out, encoder_mask
             )
             predictions.append(output.squeeze(1))
-            # If we have a target (training) then the next input is the gold
-            # symbol for this step (i.e., teacher forcing).
-            if target is not None:
+            # In teacher forcing mode the next input is the gold symbol
+            # for this step.
+            if teacher_forcing:
```

Comment:
Why can't this switch on whether or not `target` is null?

Reply:
See my email: we can still use the target for early stopping if provided, even when evaluating without teacher forcing. Is that too tricky/confusing?